### PR TITLE
Add "scratch"-taking functions

### PR DESCRIPTION
**Pull request by Dr-Emann**
Add functions which take a "scratch" type, which lzfse can use to avoid allocations.

### CODE REVIEW

File: lzfse/src/lib.rs [hunk 1]
```diff
@@ -50,12 +50,12 @@ pub enum Error {
 pub fn encode_buffer(input: &[u8], output: &mut [u8]) -> Result<usize, Error> {
     let out_size = unsafe {
         ffi::lzfse_encode_buffer(
-            output.as_ptr() as *mut _,
```
Comment (Dr-Emann) [inline lzfse/src/lib.rs:53]: I think this may have technically been undefined behavior, since this uses a shared reference, and mutates through it.

**Comment by citruz** [comment]
Sorry, for some reason I completely missed this PR! Will have a look soon.

**Comment by Dr-Emann** [comment]
@citruz gentle reminder about this

**Comment by citruz** [comment]
@Dr-Emann Thanks for your contribution and sorry for the delay! I added some doc strings and explanation. Let me know if what I wrote makes sense.

**Comment by Dr-Emann** [comment]
Looks good to me 👍

**Pull request merged**